components/builder-db/src/models/channel.rs

```diff
@@ -170,6 +170,8 @@ impl Channel {
                 origin_channel_packages::table
                     .inner_join(origin_channels::table.inner_join(origins::table)),
             )
+            .filter(origin_packages::origin.eq(&lcp.ident.origin))
+            .filter(origin_packages::name.eq(&lcp.ident.name))
```

Comment:
I know you're still working on this PR so don't be too concerned with this feedback, wanted to give the the cause of the functional test failures so you don't have to hunt them down - `.filter(origin_packages::name.eq(&lcp.ident.name))` - this function gets called from routes that expect to list all packages in an origin. Filtering on a package name here breaks a couple routes.